deploy.ts

```diff
@@ -58,12 +81,46 @@ const createSchemas = async (schema_names: string[]) => {
     // Remove whitespace in the JSON
     const json_no_ws = JSON.stringify(JSON.parse(json));
 
-    const promise = new Promise<void>((resolve, reject) => {
-      api.tx.schemas
-        .createSchema(json_no_ws, schemaDeploy.modelType, schemaDeploy.payloadLocation)
-        .signAndSend(signerAccountKeys, { nonce: nonce++ }, ({ status, events, dispatchError }) => {
+    if (shouldPropose) {
+      // Propose to create
+      const promise = new Promise<void>((resolve, reject) => {
+        api.tx.schemas
+          .proposeToCreateSchema(
+            json_no_ws,
+            schemaDeploy.modelType,
+            schemaDeploy.payloadLocation,
+            schemaDeploy.settings
+          )
+          .signAndSend(signerAccountKeys, { nonce: nonce++ }, ({ status, events, dispatchError }) => {
+            if (dispatchError) {
+              console.error("ERROR: ", dispatchError.toHuman());
+              console.log("Might already have a proposal with the same hash?");
+              reject();
+            } else if (status.isInBlock || status.isFinalized) {
+              const evt = eventWithSectionAndMethod(events, "council", "Proposed");
+              if (evt) {
+                const id = evt?.data[1];
+                const hash = evt?.data[2].toHex();
+                console.log("SUCCESS: " + schemaName + " schema proposed with id of " + id + " and hash of " + hash);
+              }
+              resolve();
+            }
+          });
+      });
+      promises.push(promise);
+    } else {
+      // Create directly via sudo
+      const tx = api.tx.schemas.createSchemaViaGovernance(
```

Comment:
What if it already exists? Do we need a way to dedup schemas before uploading?

Reply:
We don't yet in this utility. There is a `read` tool that will figure it out, but deploy does not check it yet. Not really sure it should unless it is mainnet, but deploying to mainnet is always scary.